python/common/org/python/stdlib/datetime/TimeDelta.java

```diff
@@ -0,0 +1,347 @@
+        }
+
+        TimeDelta otherObject = (TimeDelta)other;
+
+        return new TimeDelta(
+            ((Int)days).value + ((Int)otherObject.days).value,
+            ((Int)seconds).value + ((Int)otherObject.seconds).value,
+            ((Int)microseconds).value + ((Int)otherObject.microseconds).value
+        );
+    }
+
+    @org.python.Method(__doc__ = "", args = { "other" })
+    public org.python.Object __sub__(org.python.Object other) {
+        if (!(other instanceof TimeDelta)) {
+            throw new TypeError("unsupported operand type(s) for -: 'datetime.timedelta' and '" + other.typeName() + "'");
+        }
+
+        TimeDelta otherObject = (TimeDelta)other;
+
+        return new TimeDelta(
+            ((Int)days).value - ((Int)otherObject.days).value,
+            ((Int)seconds).value - ((Int)otherObject.seconds).value,
+            ((Int)microseconds).value - ((Int)otherObject.microseconds).value
+        );
+    }
+
+    @org.python.Method(__doc__ = "", args = { "other" })
+    public org.python.Object __mul__(org.python.Object other) {
+        if (other instanceof Int) {
+            Int otherInt = (Int)other;
+
+            return new TimeDelta(
+                ((Int)days).value * otherInt.value,
+                ((Int)seconds).value * otherInt.value,
+                ((Int)microseconds).value * otherInt.value
+            );
+        }
+
+        if (other instanceof Float) {
+            Double multiplicand = ((Float)other).value;
+
+            double days = multiplicand * ((Int)this.days).value;
+            double seconds = multiplicand * ((Int)this.seconds).value;
+            double microseconds = multiplicand * ((Int)this.microseconds).value;
+
+            // Make sure any fractions trickle down
+            seconds += 86400 * (days % 1);
+            microseconds += 1_000_000 * (seconds % 1);
+
+            return new TimeDelta(
+                (long)days,
+                (long)seconds,
+                (long)microseconds
+            );
+        }
+
+
+        throw new TypeError("unsupported operand type(s) for *: 'datetime.timedelta' and '" + other.typeName() + "'");
+    }
+
+    @org.python.Method(__doc__ = "", args = { "other" })
+    public org.python.Object __eq__(org.python.Object other) {
+        if (!(other instanceof TimeDelta)) {
+            return Bool.getBool(false);
+        }
+
+        TimeDelta otherObject = (org.python.stdlib.datetime.TimeDelta) other;
+        return Bool.getBool(
+            this.days.equals(otherObject.days) &&
+            this.seconds.equals(otherObject.seconds) &&
+            this.microseconds.equals(otherObject.microseconds)
+        );
+    }
+
+    @org.python.Method(__doc__ = "", args = { "other" })
+    public org.python.Object __ne__(org.python.Object other) {
+        // By definition, not equals is the reverse of equals
+        Bool equals = (Bool)__eq__(other);
+        return Bool.getBool(!equals.value);
```

Comment:
I like doing this for the comparison operators. It's simple, logical, and easy to read.